internal/cache/cache.go

```diff
@@ -0,0 +1,121 @@
+package cache
+
+import (
+	"sync"
+
+	"github.com/rudderlabs/rudder-control-plane-sdk/modelv2"
+	"github.com/rudderlabs/rudder-control-plane-sdk/notifications"
+)
+
+type WorkspaceConfigCache struct {
+	configs        *modelv2.WorkspaceConfigs
+	updateLock     sync.Mutex
+	subscribers    []*Subscriber
+	subscriberLock sync.Mutex
+}
+
+type WorkspaceConfigNotification struct{}
+
+// Get returns a copy of the current workspace configs.
+// This is in order to avoid race conditions when consumers read the configs while they are being updated.
+func (c *WorkspaceConfigCache) Get() *modelv2.WorkspaceConfigs {
+	c.updateLock.Lock()
+	defer c.updateLock.Unlock()
```

Comment:
We can use read locks here, since we are just reading from the map.
```suggestion
	c.updateLock.RLock()
	defer c.updateLock.RUnlock()
```